examples/mui-base-with-tailwind-css/src/Slider.tsx

```diff
@@ -0,0 +1,46 @@
+import * as React from 'react';
+import SliderUnstyled, {
+  SliderUnstyledThumbSlotProps,
+  SliderUnstyledProps,
+} from '@mui/base/SliderUnstyled';
+
+const Thumb = React.forwardRef(function Thumb(
+  props: SliderUnstyledThumbSlotProps,
+  ref: React.ForwardedRef<HTMLSpanElement>,
+) {
+  const { ownerState, className, children, ...other } = props;
+  return (
+    <span
+      className={`${className} ring-cyan-500 dark:ring-cyan-400 ring-2 w-4 h-4 -mt-1 -ml-2 flex items-center justify-center bg-white rounded-full shadow absolute`}
+      {...other}
+      ref={ref}
+    >
+      <span className="w-1.5 h-1.5 bg-cyan-500 dark:bg-cyan-400 rounded-full ring-1 ring-inset ring-slate-900/5" />
+      {children}
+    </span>
+  );
+});
+
+const Slider = React.forwardRef(function Slider(
+  props: SliderUnstyledProps,
+  ref: React.ForwardedRef<HTMLSpanElement>,
+) {
+  return (
+    <SliderUnstyled
+      {...props}
```

Comment:
There's an inconsistency in how additional props are spread. `Button` applies them last, while `Slider` does it first.
I think the Slider way is better in custom-built design systems. You likely don't want the application devs to override the inner workings of a component.

Reply:
This reminded me, we probably want to introduce util for merging `componentsProps` now that they allow callbacks too. I cannot just spread as before. I am saying this because, developers may provide additional `componentsProps` here on the custom `Slider` and we would basically override them. Maybe we don't need to do this in this guide immediately, we can update later, but it shows that we are missing something in terms of having a great DX.